src/collections/utils.ts

```diff
@@ -215,4 +216,43 @@ export async function dropNamespace(httpClient: HTTPClient, name: string) {
   });
   handleIfErrorResponse(response, data);
   return response;
+}
+
+export function setDefaultIdForUpsert(command: Record<string, any>, replace?: boolean) {
+  if (command.filter == null || command.options == null) {
+    return;
+  }
+  if (!command.options.upsert) {
+    return;
+  }
+  if ('_id' in command.filter) {
+    return;
+  }
+
+  if (replace) {
+    if (command.replacement != null && '_id' in command.replacement) {
+      return;
+    }
+    command.replacement._id = new ObjectId();
+  } else {
+    if (command.update != null && _updateHasKey(command.update, '_id')) {
+      return;
+    }
+    if (command.update == null) {
+      command.update = {};
+    }
+    if (command.update.$setOnInsert == null) {
+      command.update.$setOnInsert = {};
+    }
+    command.update.$setOnInsert._id = new ObjectId();
```

Comment:
Should we check if `$setOnInsert._id` is already present in the `update`, so we can use that instead of generating one?

Reply:
@kathirsvn good point. Fixed in https://github.com/stargate/stargate-mongoose/pull/102/commits/f03f2eff3e3a6f2c399569550bece7143205fc4e.